add fake request

StrongMonkey opened the pull request.

Commit:
add check timestamp logic

Commit:
set the right pointer

Commit:
rename errors

Commit:
correct fmt err

Commit:
correct fmt errs

ibuildthecloud committed:
Merge pull request #19 from StrongMonkey/rename-error

Rename error

ibuildthecloud committed:
Merge pull request #20 from StrongMonkey/add-check-ts

Add check ts

Commit:
clean up logs

Commit:
change eventID to resourceID

ibuildthecloud committed:
Merge pull request #21 from StrongMonkey/change-logs

clean up logs

Commit:
add externalId for removed containers

Commit:
correct fmt errs

Commit:
remove dockerImage and use imageUUID

Commit:
fix errors

Commit:
change test case

Commit:
change to instance imageUUID

ibuildthecloud committed:
Merge pull request #22 from StrongMonkey/add-externalId

add externalId for removed containers

ibuildthecloud committed:
Merge pull request #24 from StrongMonkey/possible-dockerimage-remove

remove dockerImage and use imageUUID

Commit:
remove ping logs

Commit:
fix image activate bugs

Commit:
add rancher_agent_image labels

Commit:
change test cases

ibuildthecloud committed:
Merge pull request #25 from StrongMonkey/remove-ping-logs

remove ping logs

ibuildthecloud committed:
Merge pull request #26 from StrongMonkey/add-image-test

fix image activate bugs

Commit:
change to agent_image labels

Commit:
clean up logs

Commit:
change duplicate naming conflict

ibuildthecloud committed:
Merge pull request #30 from StrongMonkey/clean-logs

clean up logs

ibuildthecloud committed:
Merge pull request #28 from StrongMonkey/add-rancher-image

add rancher_agent_image labels

ibuildthecloud committed:
Merge pull request #31 from StrongMonkey/change-uuid-name

change duplicate naming conflict

Commit:
update dependency

Commit:
change client to v2

Commit:
change test cases

Commit:
fix naming bug

ibuildthecloud committed:
Merge pull request #32 from StrongMonkey/change-client-v2

Change client v2

ibuildthecloud committed:
Merge pull request #33 from StrongMonkey/fix-naming-bugs

fix naming bug

Commit:
update dependency

Commit:
delete previous names

Commit:
remove cadvisor

Commit:
update dependency

Commit:
change naming logic

ibuildthecloud committed:
Set DockerUUID like PhysicalHostUUID

Commit:
change test cases

Commit:
fix config update error handling

ibuildthecloud committed:
Merge pull request #39 from ibuildthecloud/docker-uuid

Set DockerUUID like PhysicalHostUUID

ibuildthecloud committed:
Merge pull request #34 from StrongMonkey/delete-previous-names

Delete previous names

Commit:
go fmt

ibuildthecloud committed:
Merge pull request #40 from StrongMonkey/config-update

fix config update error handling

ibuildthecloud committed:
Merge pull request #38 from StrongMonkey/change-naming-logic

change naming logic

Commit:
fix build errors

Commit:
go fmt

ibuildthecloud committed:
Update drone build to test docker versions

ibuildthecloud committed:
Merge pull request #37 from StrongMonkey/no-cadvisor

No cadvisor

ibuildthecloud committed:
Merge pull request #41 from ibuildthecloud/drone

Update drone format

ibuildthecloud committed:
Shrink package a bit

ibuildthecloud committed:
Merge pull request #44 from ibuildthecloud/shrink

Shrink package a bit

ibuildthecloud committed:
Package put agent in bin folder

ibuildthecloud committed:
Merge pull request #45 from ibuildthecloud/fix-packaging

Package put agent in bin folder

Commit:
fix bugs

Commit:
add parent reply

Commit:
change test cases

ibuildthecloud committed:
Merge pull request #46 from StrongMonkey/minor-fixed

fix bugs

ibuildthecloud committed:
Send error output for delegate.request

ibuildthecloud committed:
Merge pull request #47 from ibuildthecloud/delegate

Send error output for delegate.request

Commit:
fix ipsec

ibuildthecloud committed:
Merge pull request #48 from StrongMonkey/network

fix ipsec

Commit:
fix pulling private image

ibuildthecloud committed:
Merge pull request #49 from StrongMonkey/image

fix pulling private image

Commit:
suppress ping log

Commit:
clean logs

Commit:
remove legacy code

Commit:
Support for millCpuReservation & memoryReservation

Commit:
add version ability

ibuildthecloud committed:
Merge pull request #50 from StrongMonkey/ping

suppress ping log

ibuildthecloud committed:
Merge pull request #51 from StrongMonkey/storage-fix

remove legacy code

Commit:
Add resource override env vars

Commit:
close resource and add client timeout

ibuildthecloud committed:
Merge pull request #52 from cjellick/scheduler-fields

Add fields needed for scheduler

ibuildthecloud committed:
Merge pull request #54 from StrongMonkey/leak

close resource and add client timeout

Commit:
add test for new fields

Commit:
add timeout for goroutine and http client

Commit:
Fix host labels from env var

ibuildthecloud committed:
Merge pull request #57 from StrongMonkey/solve-ping

add timeout for goroutine and http client

ibuildthecloud committed:
Merge pull request #59 from cjellick/host-label-env-var-fix

Fix host labels from env var

ibuildthecloud committed:
Merge pull request #53 from StrongMonkey/get-version

add version ability

Commit:
fix telemetry panic

ibuildthecloud committed:
Merge pull request #60 from StrongMonkey/tel-panic

fix telemetry panic

ibuildthecloud committed:
Merge pull request #56 from StrongMonkey/new-fields

Add test for new fields

Commit:
change test case

ibuildthecloud committed:
Merge pull request #61 from StrongMonkey/change-stupid-memory-fields

change test case

Commit:
update dependency

Commit:
add windows support

Commit:
add fields before set dns

Commit:
add timeout to all container list

ibuildthecloud committed:
Merge pull request #63 from StrongMonkey/fix-dns

add fields before set dns

ibuildthecloud committed:
Merge pull request #64 from StrongMonkey/add-timeout-to-all-docker-ps

add timeout to all container list

Commit:
fix dns problem

Commit:
enforce timeout to docker client except for storage one

Commit:
add backup method for fqdn

ibuildthecloud committed:
Handle volume errors on start

ibuildthecloud committed:
Merge pull request #65 from StrongMonkey/fix-dns-again

fix dns problem

Commit:
update dependency

ibuildthecloud committed:
Merge pull request #68 from ibuildthecloud/fixes

Handle volume errors on start

ibuildthecloud committed:
Merge pull request #67 from StrongMonkey/fqdn-coreos

add backup method for fqdn

ibuildthecloud committed:
Merge pull request #66 from StrongMonkey/timeout-again

enforce timeout to docker client except for storage one

Commit:
bump to 1.7.3

Commit:
clean log

Commit:
seperate test cases

ibuildthecloud committed:
Merge pull request #69 from StrongMonkey/go1.7

bump to 1.7.3

ibuildthecloud committed:
Merge pull request #71 from StrongMonkey/seperate-test

seperate test cases

Commit:
Merge pull request #70 from StrongMonkey/clean-log

clean log

Commit:
update vendor

Commit:
update engine-api to docker/docker/client

Commit:
Merge pull request #62 from StrongMonkey/windows-demo

Windows support

ibuildthecloud committed:
Merge pull request #72 from StrongMonkey/new-client

New client

ibuildthecloud committed:
Fixes for new networking

ibuildthecloud committed:
Remove test_instance_activate_links_no_service

ibuildthecloud committed:
Merge pull request #73 from ibuildthecloud/networking

Fixes for new networking

ibuildthecloud committed:
Delete unneeded code in new networking design

ibuildthecloud committed:
Add CNI support

ibuildthecloud committed:
Merge pull request #75 from ibuildthecloud/new-net

Delete unneeded code in new networking design

ibuildthecloud committed:
Assign CNI network name to label

Commit:
skip image pull variants

Commit:
Remove rancher/agent from ping test

ibuildthecloud committed:
Merge pull request #76 from ibuildthecloud/network2

Assign CNI network name to label

Commit:
add support for user namespace

Commit:
Merge pull request #77 from cjellick/skip-pull-variants

Fix or skip slow tests

Commit:
remove parse logic in agent

Commit:
Merge pull request #80 from StrongMonkey/image-fix

remove parse logic in agent

Commit:
Merge pull request #78 from StrongMonkey/user-namespace

add support for user namespace

Commit:
update vendor

Commit:
update getIP method

Commit:
delete testPrivateImage

Commit:
Merge pull request #82 from StrongMonkey/delete-test

delete testPrivateImage

Commit:
Test fixes

Commit:
Merge pull request #83 from cjellick/test-fixes

Test fixes

ibuildthecloud committed:
Fix wrong entrypoint insertion

ibuildthecloud committed:
Merge pull request #84 from ibuildthecloud/fix-entry

Fix wrong entrypoint insertion

ibuildthecloud committed:
Merge pull request #81 from StrongMonkey/new-ip

New getIP

ibuildthecloud committed:
When running net:none still return exposed ports

ibuildthecloud committed:
Merge pull request #85 from ibuildthecloud/fix-ports

When running net:none still return exposed ports

ibuildthecloud committed:
Get pid from docker

ibuildthecloud committed:
Merge pull request #86 from ibuildthecloud/fix

Get pid from docker

Commit:
image noop fix

ibuildthecloud committed:
Fix memory leaks

Both http.Client and NsHandle usage was causing FDs to leak.
Additionally the error handling for getIP was cleaned up

ibuildthecloud committed:
Merge pull request #89 from ibuildthecloud/fixes2

Fix memory leaks

Commit:
don't return error if noop

Commit:
short life container ip fix

ibuildthecloud committed:
Merge pull request #88 from StrongMonkey/fix

image noop fix

ibuildthecloud committed:
Merge pull request #90 from StrongMonkey/noop-fix

don't return error if noop

ibuildthecloud committed:
Support docker 1.10 and 1.11

ibuildthecloud committed:
Merge pull request #92 from ibuildthecloud/support-10-11

Support docker 1.10 and 1.11

ibuildthecloud committed:
Merge pull request #91 from StrongMonkey/fix1

short life container ip fix

ibuildthecloud committed:
Delete unneeded code

Commit:
Remove volume driver

Commit:
fix

Commit:
Merge pull request #95 from StrongMonkey/fix2

fix

Commit:
Merge pull request #94 from cjellick/vol-driver-fix

Remove volume driver

Commit:
Merge pull request #93 from ibuildthecloud/unneeded

Delete unneeded code

Commit:
new golint style

Commit:
add support for windows ci

dx9 committed:
Bring -static back

ibuildthecloud committed:
Merge pull request #96 from StrongMonkey/new-golint

new golint style

ibuildthecloud committed:
Revert "fix"

This reverts commit ff0f4ed1f5f437870df582b9133c76506605a49b.

ibuildthecloud committed:
Fix timeout on IP

ibuildthecloud committed:
Merge pull request #98 from dx9/static-link

Bring -static back

ibuildthecloud committed:
Merge pull request #100 from ibuildthecloud/fix2

Fix2

Commit:
delete unneeded code

Commit:
change uuid to container name

ibuildthecloud committed:
Merge pull request #102 from StrongMonkey/container-network

change uuid to container name

Commit:
Merge pull request #101 from StrongMonkey/delete

delete unneeded code

Commit:
update vendor

Commit:
update .gitignore and scripts

Commit:
migrating host-api

Commit:
windows support

Commit:
Merge pull request #97 from StrongMonkey/windows-ci

add support for windows ci

Commit:
Merge pull request #87 from StrongMonkey/add-host-api

Migrating host-api and add windows supports

Commit:
reflect change

Commit:
update vendor

Commit:
Merge pull request #104 from StrongMonkey/master

rancher/rancher#6957

Commit:
fix race condition

Commit:
don't use timeout for dockerclient when pulling images

Commit:
don't panic just pass as python-agent did

Commit:
use both cases for proxy envs

Commit:
env fix

Commit:
Do build during package if binary doesnt exist

ibuildthecloud committed:
Merge pull request #109 from StrongMonkey/envfix

env fix

ibuildthecloud committed:
Merge pull request #108 from StrongMonkey/proxy

use both cases for proxy envs

ibuildthecloud committed:
Merge pull request #107 from StrongMonkey/rancher6910

don't panic just pass as python-agent did

ibuildthecloud committed:
Merge pull request #106 from StrongMonkey/imagefix

don't use timeout for dockerclient when pulling images

ibuildthecloud committed:
Merge pull request #105 from StrongMonkey/racefix

fix race condition

ibuildthecloud committed:
Support non-/var/lib/docker root

ibuildthecloud committed:
Merge pull request #111 from ibuildthecloud/non-standard-root

Support non-/var/lib/docker root

Commit:
don't use hard code for image authConfig

ibuildthecloud committed:
Add service DNS first

ibuildthecloud committed:
Merge pull request #112 from StrongMonkey/auth-fix

don't use hard code for image authConfig

ibuildthecloud committed:
Merge pull request #113 from ibuildthecloud/dns

Add service DNS first

Commit:
set proxy only for system container and override it

Commit:
Merge pull request #110 from cjellick/fix-package-script

Do build during package if binary doesnt exist

Commit:
populate io.rancher.scheduler.ips label in each host

Commit:
Merge pull request #114 from StrongMonkey/proxy-fix

set proxy only for system container and override it

Commit:
add panic file log

Commit:
Merge pull request #115 from StrongMonkey/add-panic

add panic file log

ibuildthecloud committed:
Don't use /var/lib/docker anymore

ibuildthecloud committed:
Merge pull request #116 from ibuildthecloud/no-var-lib-docker

Don't use /var/lib/docker anymore

Commit:
Disable cgo on build

Commit:
don't use strong type for DockerInspect.State

ibuildthecloud committed:
Merge pull request #120 from StrongMonkey/healthcheck-fix

don't use strong type for DockerInspect.State

Commit:
Add mac address as a label

Commit:
Merge pull request #121 from cjellick/add-mac-label

Add mac address as a label

Commit:
Dont use /dev/sda in tests

Commit:
Add mutex to protect against race condition

Commit:
Merge pull request #123 from cjellick/racefix

Add mutex to protect against race condition

ibuildthecloud committed:
Merge pull request #122 from cjellick/dont-use-dev-sda

Dont use /dev/sda in tests

imikushin committed:
Attach rancher storage volumes to the host before starting container

Commit:
Check state file for CNI IP and error

Commit:
add fake request

StrongMonkey commented:
https://github.com/rancher/rancher/issues/7298

ibuildthecloud commented on handlers/config_update.go:
```diff
@@ -9,12 +9,36 @@ import (
+		items := []string{"host-config", "pyagent", "configscripts"}
+		urls := []string{}
+		for _, item := range items {
+			url := os.Getenv("CATTLE_URL") + fmt.Sprintf("/configcontent/%v?version=latest", item)
```
Can you use `Config.APIURL("")` and `Config.AccessKey()/SecretKey()`

StrongMonkey replied:
yeah i ll change it to that

Commit:
don't lose decimal for flt

ibuildthecloud committed:
Merge pull request #124 from cjellick/cni-state-file

Check state file for CNI IP and error

ibuildthecloud committed:
Merge pull request #119 from cjellick/disable-cgo

Disable cgo on build

Commit:
Merge pull request #128 from StrongMonkey/host-stats-fix

don't lose decimal for float number

Commit:
Revert "don't lose decimal for float number"

Commit:
Merge pull request #129 from rancher/revert-128-host-stats-fix

Revert "don't lose decimal for float number"

Commit:
Close logs when user cancels request

Commit:
Don't use go routine for exec reader

The combination of the go routine and select {} was orphaning the go
routine that contained the select {}.

Commit:
try three times then return success

Commit:
Merge pull request #131 from StrongMonkey/volume-fix

try three times then return success

ibuildthecloud committed:
Merge pull request #130 from cjellick/kill-logs

Close logs when user cancels request

Commit:
small fix

ibuildthecloud commented on handlers/config_update.go:
```diff
@@ -7,14 +7,38 @@ import (
 func (h *ConfigUpdateHandler) ConfigUpdate(event *revents.Event, cli *client.RancherClient) error {
+	if runtime.GOOS == "windows" {
+		// if os is windows, do a fake reply and manually touch some urls as @ibuildthecloud suggested
+		items := []string{"host-config", "pyagent", "configscripts"}
```
These items should be in the event that is sent.  It will tell you what items to update.

ibuildthecloud committed:
Merge pull request #132 from StrongMonkey/fix-2

small fix

ibuildthecloud committed:
Generate and send host rsa key

Commit:
Merge pull request #118 from StrongMonkey/scheduler

Populate io.rancher.host.ip_set label in each host

ibuildthecloud committed:
Merge pull request #133 from ibuildthecloud/key

Generate and send host rsa key

ibuildthecloud committed:
Merge pull request #127 from imikushin/attach

Attach managed volumes to the host before starting container

Commit:
fix stats bugs

cloudnautique committed:
Switch to dir instead of base

Commit:
Merge pull request #135 from cloudnautique/fix_agent_issues

Switch to dir instead of base

Commit:
fix image pull log

Commit:
bump to 1.7.5

Commit:
fix agent leak

Commit:
Merge pull request #138 from StrongMonkey/agent-leak

fix agent leak

Commit:
don't refresh ip label

Commit:
Merge pull request #140 from StrongMonkey/no-refresh

don't refresh ip label

ibuildthecloud committed:
Add 1.13.1 matrix build

ibuildthecloud committed:
Merge pull request #141 from ibuildthecloud/113

Add 1.13.1 matrix build

Commit:
Merge pull request #136 from StrongMonkey/storage-log

fix image pull log

Commit:
Merge pull request #137 from StrongMonkey/go1.7.5

bump to 1.7.5

Commit:
don't use timeout client in instance.stop

Commit:
Merge pull request #134 from StrongMonkey/stats-fix

fix stats bugs

Commit:
Merge pull request #142 from StrongMonkey/timeout-stop

don't use timeout client in instance.stop

StrongMonkey commented:
This PR is updated to address @ibuildthecloud 's code review.
